src-v2/Judge/NTS.Judge/Adapters/Behinds/ParticipationBehind.cs

```diff
@@ -115,6 +120,8 @@ public async Task RevokeQualification(int number, QualificationRevokeType type,
             participation.FailToCompleteLoop(reason);
         }
         await _participationRepository.Update(participation);
+
+        EmitChange();
     }
```

Comment:
This is one big ass elimination method. Why don't we get rid of the last if-chain and split this up into named methods following the domain model?